kani-driver/src/args.rs

```diff
@@ -135,19 +135,9 @@ pub struct KaniArgs {
     #[arg(long, hide_short_help = true)]
     pub only_codegen: bool,
 
-    /// Disable the new MIR Linker. Using this option may result in missing symbols from the
-    /// `std` library. See <https://github.com/model-checking/kani/issues/1213> for more details.
+    /// Enable the new MIR Linker. This is a no-op since we no longer support any other linker.
     #[arg(long, hide = true)]
-    pub legacy_linker: bool,
```

Comment:
I'm fine with removing it outright, but as a pattern for future removals, maybe we should leave the flag but issue warnings/errors (i.e. a good error message is better than a bad one. "This flag no longer does anything, remove it" is better than "unknown flag")

But I suspect no one is using this one but us. Still, maybe we should add the warning (or error, probably) for `--mir-linker` now?

Reply:
Sure. I can do the same for the legacy linker too. I debated whether to do that or not, but it's probably a good habit anyways.